compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -1293,9 +1296,20 @@ impl<'a, G: EmissionGuarantee> Diag<'a, G> {
     /// `cancel`, etc. Afterwards, `drop` is the only code that will be run on
     /// `self`.
     fn take_diag(&mut self) -> DiagInner {
+        if let Some(path) = &self.long_ty_path {
+            self.note(format!(
+                "the full name for the type has been written to '{}'",
+                path.display()
```

Comment:
Question: not necessarily for this PR, but this doesn't handle path-remapping (`--remap-path-prefix` or `-Zremap-path-scope=diagnostics` and such), right?

Reply:
Correct. We should change it to do so.